File: RxDiRetrofitApp/japp/src/main/java/com/woowrale/rxdiretrofit/data/repository/ApiServices.java
package com.woowrale.rxdiretrofit.data.repository;


import com.woowrale.rxdiretrofit.data.model.Crypto;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface ApiServices {

    @GET("{coin}-usd")
    Observable<Crypto> getCoinData(@Path("coin") String coin);
}

File: AndroidPatternsApp/kmvp-app/src/main/java/com/woowrale/mvp/data/repository/ApiServices.kt
package com.woowrale.mvp.data.repository

import com.woowrale.mvp.data.model.Contact
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.Url


interface ApiContacts {

    @GET
    fun getContacts(
        @Url apiContacts: String,
        @Query("source") source: String,
        @Query("search") query: String
    ): Single<List<Contact>>

}